### ISSUE TITLE
 ServiceStack.Redis RedisSentinel fails to initialise when specifying initial db other than 0

### ISSUE THREAD

**Issue by oclancy** [ServiceStack/Issues#216]
If I try and initialise a RedisSentinel with a connection string with a db set to anything other than 0, the RedisSentinel fails to initialise. The error message is "Unable to resolve sentinels!". There is no other logging information. The connection string is "127.0.0.1:26379?db=1".

e.g.  var tempSentinel = new RedisSentinel([]{"127.0.0.1:6379?db=1"}, "mymaster");

will fail with a RedisException.
 
As I say a connection string of "127.0.0.1:26379?db=0" is successful.

I am using version ServiceStack.Redis 4.0.35 on .Net 4.5

**Comment by mythz** [comment]
This is because the connection string provided is to the configure Redis Sentinel connection which is a limited version of `redis-server` that doesn't support Database operations.

You can achieve what you're after by specifying the initial DB with the `PooledRedisClientManager` that's used to create the `RedisClient` connections, e.g:

```csharp
var sentinel = new RedisSentinel([]{"127.0.0.1:6379"}, "mymaster") {
    RedisManagerFactory =
    {
        FactoryFn = (writeHosts, readHosts) =>
            new PooledRedisClientManager(writeHosts, readHosts, initalDb:1)
    }
};

using (var clientsManager = (PooledRedisClientManager)sentinel.Setup())
using (var client = clientsManager.GetClient())
{
    Assert.That(client.Db, Is.EqualTo(1));
}
```

I'll look to see if there's an easy way to propagate the DB configuration down to the Redis Client Managers used by RedisSentinel.

**Comment by mythz** [comment]
Hi,

The nicest API I could get this down to is to use a new `HostFilter` which provides a way to be able to customize the host connection string before it gets passed into the underlying ClientsManager, e.g:

```csharp
var sentinel = new RedisSentinel([]{"127.0.0.1:6379"}, "mymaster") {
    HostFilter = host => host + "?db=1"
};
```

This keeps the connection string between the **redis-sentinel** and the **redis-server** appropriately distinct but still allows you to configure the **redis-server** clients with a connection string.

This change is available from v4.0.36+ that's [now available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).
